### PR TITLE
Disable fade in for monetizaton prompts called from rewarded ads

### PR DESCRIPTION
**Pull request by mhkawano**
Disable the fade in animation for the graypane when using rewarded ads so that it does not flicker when switching between them.

b/301934795

### CODE REVIEW

File: src/api/subscriptions.ts [hunk 1]
```diff
@@ -512,6 +512,11 @@ export interface OffersRequest {
    * upgrade or downgrade their current subscription.
    */
   oldSku?: string;
+
+  /**
+   * Optional. Sets if the fade in of is animated.
+   */
+  shouldNotAnimateFade?: boolean;
```
Comment (justinchou-google) [inline src/api/subscriptions.ts:519]: can we make them all `shouldAnimateFade` for consistency?

Comment (mhkawano) [reply src/api/subscriptions.ts]: Done.

Comment (mhkawano) [reply src/api/subscriptions.ts:519]: I set them all to `shouldNotAnimateFade` so that if the value is unset, it will default to animating to preserve current behavior.

File: src/api/subscriptions.ts [hunk 2]
```diff
@@ -512,6 +512,11 @@ export interface OffersRequest {
    * upgrade or downgrade their current subscription.
    */
   oldSku?: string;
+
+  /**
+   * Optional. Sets if the fade in of is animated.
```
Comment (fikus) [inline src/api/subscriptions.ts:517]: nit: check the wording of this sentence?

File: src/runtime/auto-prompt-manager.ts [hunk 3]
```diff
@@ -766,7 +767,8 @@ export class AutoPromptManager {
               isClosable,
               monetizationFunction: this.getLargeMonetizationPromptFn_(
                 autoPromptType,
-                !!isClosable
+                !!isClosable,
+                /* shouldNotAnimateFade */ true
```
Comment (fikus) [inline src/runtime/auto-prompt-manager.ts:771]: Read from `options`?

File: src/api/subscriptions.ts (hunk 1 above)
Comment (fikus) [reply src/api/subscriptions.ts:519]: I agree, the logic is hard to follow with the double negations.

Also consider that this is updating the public API. Is this necessary? Can we pass this option in another way?

If we do need to update the API, it's important that we keep it readable. IMO it would be better to name this positively with a comment that it defaults to true, and implement accordingly.

File: src/runtime/auto-prompt-manager.ts (hunk 3 above)
Comment (mhkawano) [reply src/runtime/auto-prompt-manager.ts:771]: This is for rewarded ads, so it should always suppress the animation here.

File: src/api/subscriptions.ts (hunk 1 above)
Comment (fikus) [reply src/api/subscriptions.ts:519]: If the current implementation requires updating the API, can you add a TODO to revisit? This UI detail doesn't really belong in OffersRequest.

Comment (mhkawano) [reply src/api/subscriptions.ts]: Done. Updated the comment and filed a bug to clean this up later.

File: src/api/subscriptions.ts [hunk 4]
```diff
@@ -512,6 +512,13 @@ export interface OffersRequest {
    * upgrade or downgrade their current subscription.
    */
   oldSku?: string;
+
+  /**
+   * Optional. Disables the fade in animation if set to false. Defaults to true
+   * if unset.
+   * TODO: b/304803271 - remove this field form the api.
```
Comment (fikus) [inline src/api/subscriptions.ts:519]: nit: typo "form"

**Review by fikus — APPROVED** [review]
